plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java

```diff
@@ -331,6 +332,47 @@ public void testSqlServerNchar()
                 .execute(getQueryRunner(), sqlServerCreateAndInsert("test_sqlserver_char"));
     }
 
+    @Test
+    public void testSqlServerUniqueIdentifier()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("uniqueidentifier", "NULL", UUID, "CAST(NULL AS uuid)")
+                .addRoundTrip("uniqueidentifier", "'d5416a8a-5faf-208d-a0cc-aaf97732f49a'", UUID, "uuid 'd5416a8a-5faf-208d-a0cc-aaf97732f49a'")
+                .addRoundTrip("uniqueidentifier", "'00000000-0000-0000-0000-000000000000'", UUID, "uuid '00000000-0000-0000-0000-000000000000'")
+                .addRoundTrip("uniqueidentifier", "'ffffffff-ffff-ffff-ffff-ffffffffffff'", UUID, "uuid 'ffffffff-ffff-ffff-ffff-ffffffffffff'")
+                .addRoundTrip("uniqueidentifier", "'0E984725-C51C-4BF4-9960-E1C80E27ABA0wrong'", UUID, "uuid '0E984725-C51C-4BF4-9960-E1C80E27ABA0'")
+                .execute(getQueryRunner(), sqlServerCreateAndInsert("test_sqlserver_uniqueidentifier"));
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("uniqueidentifier", "NULL", UUID, "CAST(NULL AS uuid)")
+                .addRoundTrip("uniqueidentifier", "uuid 'd5416a8a-5faf-208d-a0cc-aaf97732f49a'", UUID, "uuid 'd5416a8a-5faf-208d-a0cc-aaf97732f49a'")
+                .addRoundTrip("uniqueidentifier", "uuid '00000000-0000-0000-0000-000000000000'", UUID, "uuid '00000000-0000-0000-0000-000000000000'")
+                .addRoundTrip("uniqueidentifier", "uuid 'ffffffff-ffff-ffff-ffff-ffffffffffff'", UUID, "uuid 'ffffffff-ffff-ffff-ffff-ffffffffffff'")
+                .execute(getQueryRunner(), sqlServerCreateAndTrinoInsert("test_sqlserver_uniqueidentifier"));
+    }
+
+    @Test
+    public void testUnsupportedUniqueIdentifier()
+    {
+        try (TestTable table = new TestTable(onRemoteDatabase(), "test_unsupported_uniqueidentifier", "(data uniqueidentifier)")) {
+            assertSqlServerQueryFails(
+                    "INSERT INTO " + table.getName() + " VALUES ('0E984725-EXTRA-4BF4-9960-E1C80E27ABA0')", // Invalid uniqueidentifier
```

Comment:
The comment isn't helpful.
```suggestion
                    "INSERT INTO " + table.getName() + " VALUES ('0E984725-EXTRA-4BF4-9960-E1C80E27ABA0')",
```

Reply:
Will `// UUID should match regex: /^[0-9A-F]{8}-[0-9A-F]{4}-[4][0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/i` be clear for this?

Reply:
Ah I just saw you have suggested a change. I will adopt it then. Thanks!